### DISCUSSION TITLE
Keep default value for `shouldUnregister: true` in `useController`

### DISCUSSION THREAD

**Discussion by darren-fp** [react-hook-form/react-hook-form#8409] [Q&A]
Hey folks, thanks for the amazing library! Based on the [documentation](https://react-hook-form.com/api/usecontroller), when `shouldUnregister` is`true` in `useController`, input will be unregistered after unmount and defaultValues will be removed as well. Is it possible to configure such that defaultValues will not be removed on unmount, something like how `unregister` allows us to `keepDefaultValue`?

If possible, I would like to do something like this:
```
const methods = useController(shouldUnregister: { keepDefaultValue: true })
```

Thanks!

**Comment by bluebill1049** [comment]
you can use `unregsiter` function to control the unmount behaviour.

**Reply by darren-fp** [reply]
Is it possible to avoid using the `unregister` function? That would add extra code for unregistering the input

**Reply by Moshyfawn** [reply]
i don't think there's an API for that, as the `shouldUnregister` flag is setup to do exactly that: unregister an input.

You can use the `defaultValues` to provide a default value to your payload `onSubmit`. Something like:
```js
const onSubmit = data => {
  const payload = {
    unregisteredField: data.unregisteredField || defaultValues.unregisteredField
  }
}
```

**Reply by darren-fp** [reply]
Yes I am using `shouldUnregister` flag but it currently doesnt allow me to configure what can be kept unlike `unregister` API

I created a sandbox to illustrate the issue. In line 55, I have to write extra code to configure the unregister behaviour to keep default values. If I use the `shouldUnregister` flag in line 10 instead of useEffect in line 55, the default values will not be kept when the API returns and component remounts

https://codesandbox.io/s/keep-defaultvalue-for-shouldunregister-true-8cfkjh?file=/src/index.js

**Comment by leapful** [comment]
@yilong-tan : Could you please explain why you cannot use "shouldUnregister" option on "useForm" hook ? Something like:

```
useForm({
  // Other options
  shouldUnregister: true
})
```

Working demo: https://codesandbox.io/s/keep-defaultvalue-for-shouldunregister-true-forked-5u3m91?file=/src/index.js

**Reply by darren-fp** [reply]
Yes this seems to work but I am curious why there is a difference in behaviour between `shouldUnregister: true` for `useForm` and `useController`. It seems like default values are kept when `shouldUnregister` is true for `useForm` but not for `useController`

**Reply by darren-fp** [reply]
Also ideally we would like to set `shouldUnregister: true` in every controlled input (i.e. useController) instead of every form (i.e. useForm) so that it easier to maintain since there are only a few input types but there can be an unlimited number of forms
